extensions/packageManager/src/client/feedModal.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { PrimaryButton, DialogFooter, DefaultButton } from 'office-ui-fabric-react';
+import { useState, useEffect, Fragment } from 'react';
+import {
+  CheckboxVisibility,
+  DetailsList,
+  DetailsListLayoutMode,
+  Selection,
+  SelectionMode,
+} from 'office-ui-fabric-react/lib/DetailsList';
+import { IconButton, ActionButton } from 'office-ui-fabric-react/lib/Button';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { useApplicationApi } from '@bfc/extension-client';
+
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+import formatMessage from 'format-message';
+import { v4 as uuid } from 'uuid';
+
+import { PackageSourceFeed } from './Library';
+
+const uitext = {
+  title: formatMessage('Feeds'),
+  subTitle: formatMessage('Manage the package sources'),
+};
+
+export interface WorkingModalProps {
+  hidden: boolean;
+  title: string;
+  feeds: PackageSourceFeed[];
+  closeDialog: any;
+  onUpdateFeed: any;
+}
+export const FeedModal: React.FC<WorkingModalProps> = (props) => {
+  const [selectedItem, setSelectedItem] = useState<PackageSourceFeed | undefined>(undefined);
+  const [items, setItems] = useState<PackageSourceFeed[]>(props.feeds);
+  const [editRow, setEditRow] = useState<boolean>(false);
+  const { confirm } = useApplicationApi();
+
+  const [selection, setSelection] = useState<Selection>(
+    new Selection({
+      onSelectionChanged: () => {
+        if (selection.getSelectedCount() > 0) {
+          setSelectedItem(selection.getSelection()[0] as PackageSourceFeed);
+        } else {
+          setSelectedItem(undefined);
+        }
+      },
+    })
+  );
+
+  useEffect(() => {
+    setItems(props.feeds);
+  }, [props.feeds]);
+
+  useEffect(() => {
+    if (selectedItem && selectedItem.text === 'New Feed') {
```

Comment:
this is going to be localized, if yes, then this check will fail.

Reply:
use selectedItem.key instead and make sure the key is consistent.

Reply:
this is a tricky one - the key is a generated guid right now so no easy way to identify a new item that way. 

i'm not sure that text should be localized using formatMessage since it is the default value of an editable field not UI text.

Reply:
you can remove this effect and set editRow to true within addItem and set it to false when selection change

Reply:
good one!